packages/cpsuLambda/src/schema/format_1/transformer.ts

```diff
@@ -78,10 +79,23 @@ export function generateTemplate(requestBody: requestType): string {
 export function populateTemplate(
   template: string,
   prescriptionItem: itemType,
-  prescriptionDetails: requestType
-): Result<BundleEntry<Task>, string> {
+  prescriptionDetails: requestType,
+  _logger: Logger
+): Result<BundleEntry<Task>, string> | undefined {
   const entry = JSON.parse(template) as BundleEntry<Task>
 
+  if (prescriptionItem.status === "DispensingComplete") {
+    const forbiddenStatuses = ["Expired", "NotDispensed"]
+
+    if (prescriptionItem.completedStatus && forbiddenStatuses.includes(prescriptionItem.completedStatus)) {
+      _logger.info("Skipping data store update for DispensingComplete - completedStatus is an ignored value", {
+        itemID: prescriptionItem.itemID,
+        completedStatus: prescriptionItem.completedStatus
+      })
+      return undefined
```

Comment:
we should be using null not undefined

Reply:
I've returned an `Err`, and filtered them out when the populated templates are collected.